### PR TITLE
Update `tokio` to 0.3

### PR DESCRIPTION
**Pull request by nickelc**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,10 +33,10 @@ version = "0.2.0"
 
 [dependencies.tokio-native-tls]
 optional = true
-version = "0.1"
+version = "0.2"
 
 [dev-dependencies]
 futures-channel = "0.3"
-tokio = { version = "0.2", default-features = false, features = ["io-std", "macros", "stream", "time"] }
+tokio = { version = "0.3", default-features = false, features = ["io-std", "macros", "rt-multi-thread", "stream", "time"] }
```
Comment (c410-f3r) [inline Cargo.toml:40]: Is `rt-multi-thread` really necessary? This feature will bring more transient dependencies to libraries

Comment (nickelc) [reply Cargo.toml:40]: yes, it's required for as `dev-dependency` for the tests.

Comment (c410-f3r) [reply Cargo.toml:40]: Hahahahahaha :D

These are `dev-dependencies`. Sorry for the fuzz

Comment (nickelc) [reply Cargo.toml:40]: no problem, tokio 0.3 changed a bit in its `macros` & runtime handling
This is the error you get now
```
error: The #[tokio::test] macro requires rt or rt-multi-thread.
 --> tests/handshakes.rs:4:1
  |
4 | #[tokio::test]
  | ^^^^^^^^^^^^^^
```

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
LTGM thanks!

**Pull request merged**